### ISSUE TITLE
Cancelling a spin attack with dpad Ocarina results in the Magic Bar returning to Vanilla color during the UI fadeout

### ISSUE THREAD

**Issue by sorenbug** [TestRunnerSRL/OoT-Randomizer#1058]
To execute:
- Randomize cosmetics (or at least magic color)
- Acquire magic meter and ocarina
- Hold B so that your spin attack charges
- Without releasing B, press Dpad Down

The magic bar will revert to Vanilla coloring for the ~0.5s of UI fadeout.

**Comment by r0bd0g** [comment]
Did we change when ocarina dpad is allowed recently? I wonder if this one still occurs?

**Comment by s5bug** [comment]
Still occurs on the latest build.

**Comment by Sirius902** [comment]
Yep, I can confirm on latest `Dev` which is 20045ff3c68e79ab30fd09c017f4c10a9da01148 as of right now. Here's a video.

https://user-images.githubusercontent.com/10891979/215888722-66f86f23-d47b-4ac1-994f-f87aca53dbe7.mp4

**Comment by r0bd0g** [comment]
Does this happen with c button ocarina?
Feels like this is probably the fadeout not really know what colours to use? No idea.

**Comment by djevangelia** [comment]
This is because the alpha that gets applied, because the yellow bar is actually drawn all the way from the left where the bar starts to the right magic edge. If the yellow bar is drawn only where the magic to be used is (from magicTarget to current magic), this effect is removed.

It's fixed in decomp [here](https://github.com/zeldaret/oot/blob/55cdfccdd90f96538274bb8fe6854d1d33fa87e9/src/code/z_parameter.c#L2718) by changing `R_MAGIC_FILL_X << 2` to `(R_MAGIC_FILL_X + gSaveContext.magicTarget) << 2`.

I tried doing something in assembly but ran into issues so I won't be working on this I think but the fix is very simple.
